streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -99,6 +100,7 @@
     private final InternalProcessorContext processorContext;
     private final RecordQueueCreator recordQueueCreator;
 
+    private StampedRecord record;
```

Comment:
We pull this variable out form the method to use it as a "cache" -- if we fail on send(), we can process the record a second time.